release_bot/celery_task.py

```diff
@@ -0,0 +1,117 @@
+    """Handler for newly opened issues"""
+    configuration.configuration = Path(getenv("CONF_PATH",
+                                              "/home/release-bot/.config/conf.yaml")).resolve()
+
+    # add configuration from Github webhook
+    configuration.repository_name = webhook_payload['repository']['name']
+    configuration.repository_owner = webhook_payload['repository']['owner']['login']
+    configuration.github_username = webhook_payload['issue']['user']['login']
+    configuration.load_configuration()  # load the rest of configuration if there is any
+
+    # create url for github app to enable access over http
+    configuration.clone_url = f'https://x-access-token:' \
+        f'{configuration.github_token}@github.com/' \
+        f'{configuration.repository_owner}/{configuration.repository_name}.git'
+
+    logger = configuration.logger
+    release_bot = ReleaseBot(configuration)
+
+    logger.info("Resolving opened issue")
+    release_bot.git.pull()
+    try:
+        release_bot.load_release_conf()
+        if (release_bot.new_release.trigger_on_issue and
+                release_bot.find_open_release_issues()):
+            if release_bot.new_release.labels is not None:
+                release_bot.project.add_issue_labels(
+                    release_bot.new_pr.issue_number,
+                    release_bot.new_release.labels)
+            release_bot.make_release_pull_request()
+    except ReleaseException as exc:
+        logger.error(exc)
+
+
+def handle_pr(webhook_payload):
+    """Handler for merged PR"""
+    configuration.configuration = Path(getenv("CONF_PATH",
+                                              "/home/release-bot/.config/conf.yaml")).resolve()
+
+    # add configuration from Github webhook
+    configuration.repository_name = webhook_payload['repository']['name']
+    configuration.repository_owner = webhook_payload['repository']['owner']['login']
+    configuration.github_username = webhook_payload['pull_request']['user']['login']
+    configuration.load_configuration()  # load the rest of configuration if there is any
+
+    # create url for github app to enable access over http
+    configuration.clone_url = f'https://x-access-token:' \
+        f'{configuration.github_token}@github.com/' \
+        f'{configuration.repository_owner}/{configuration.repository_name}.git'
+
+    logger = configuration.logger
+    release_bot = ReleaseBot(configuration)
+
+    logger.info("Resolving opened PR")
+    release_bot.git.pull()
+    try:
+        release_bot.load_release_conf()
+        if release_bot.find_newest_release_pull_request():
+            release_bot.make_new_github_release()
+            # Try to do PyPi release regardless whether we just did github release
+            # for case that in previous iteration (of the 'while True' loop)
+            # we succeeded with github release, but failed with PyPi release
+            release_bot.make_new_pypi_release()
```

Comment:
We need to figure out how we will handle PyPI releases via GithubApp, but we should create a separate issue for that.